parser/parser.go

```diff
@@ -157,31 +158,44 @@ func gTableNames() ([]string, error) {
 	return tables, nil
 }
 
-func gTableParams() ([]string, error) {
+func gTableParams() ([]string, []string, error) {
```

Comment:
it's really rare to see a function in go returning 3 values. 
I wonder if it's not easier to follow this approach: 

we might keep gitql searching for all the rows and in the end, when it generates the dataTable, we filter it. So the idea would be generate a new filtered table based on [the original one](https://github.com/filhodanuvem/gitql/blob/master/runtime/runtime.go#L95). 

I have an impression that for your solution, we could write `select distinct name from branches` but the proper distinct implementation is done just for the table commits. Is my making sense?

Reply:
That's correct. `walkCommits` and `walkReferences` are nearly identical and I wasn't expecting that. It seems like the only thing that changes (from a glance) is the `metadataReference` and the `metadataCommit` calls, it looks like this could be generelized into a single `func` that would receive another `func`, the metadata extractor and avoid code duplication.

Non the less I won't change that now and I'll try your approach, seems better since we would only have to apply this logic once. The downside is that we would be creating a possibly big table that won't be used, but that isn't a pratical concern for the purpose of this tool I guess.

Reply:
This shouldn't have hitted me before, but here goes: if we filter only after we have the dataTable we will only be filtering distinct values after the limit applies, consider the following commit history:

```
commit 6: claudson ...
commit 5: claudson ...
commit 4: claudson ...
commit 3: claudson ...
commit 2: claudson ...
commit 1: someoneelse ...
```

In this PRs current implementation we get back:
```
claudson
someoneelse
```

If we wait to filter distinct values after we've constructed the data table, we get:
```
claudson
```

Because the `s.Limit` filter has already been applied inside `walkCommits` and  `walkReferences`. We can filter the table inside the `walkCommits` and `walkReferences` before the limit is applied, do you prefer that approach over the one currently implemented in the PR?

Reply:
That's a reaaaaaaally good point. It's seems better to go with your solution even supporting just commits. Let's do that and iterate later.